commit-event-service/src/main/scala/ch/datascience/commiteventservice/events/categories/commitsync/eventgeneration/CommitEventSynchronizer.scala

```diff
@@ -0,0 +1,258 @@
+    executionTimeRecorder: ExecutionTimeRecorder[Interpretation],
+    logger:                Logger[Interpretation],
+    clock:                 java.time.Clock = java.time.Clock.systemUTC()
+) extends CommitEventSynchronizer[Interpretation] {
+
+  import executionTimeRecorder._
+  override def synchronizeEvents(event: CommitSyncEvent): Interpretation[Unit] = (for {
+    maybeAccessToken  <- accessTokenFinder.findAccessToken(event.project.id)
+    maybeLatestCommit <- latestCommitFinder.findLatestCommit(event.project.id, maybeAccessToken).value
+    _                 <- checkForSkippedEvent(maybeLatestCommit, event, maybeAccessToken)
+  } yield ()) recoverWith loggingError(event)
+
+  private def checkForSkippedEvent(maybeLatestCommit: Option[CommitInfo],
+                                   event:             CommitSyncEvent,
+                                   maybeAccessToken:  Option[AccessToken]
+  ) = (maybeLatestCommit, event) match {
+    case (Some(commitInfo), FullCommitSyncEvent(id, _, _)) if commitInfo.id == id =>
+      measureExecutionTime(Skipped.pure[Interpretation].widen[UpdateResult]) >>= logResult(event)
+    case (Some(commitInfo), event) =>
+      processCommitsAndLogSummary(commitInfo.id, event.project, maybeAccessToken)
+    case (None, FullCommitSyncEvent(id, _, _)) =>
+      processCommitsAndLogSummary(id, event.project, maybeAccessToken)
+    case (None, MinimalCommitSyncEvent(_)) =>
+      measureExecutionTime(Skipped.pure[Interpretation].widen[UpdateResult]) >>= logResult(event)
+  }
+
+  private def processCommitsAndLogSummary(commitId: CommitId, project: Project, maybeAccessToken: Option[AccessToken]) =
+    measureExecutionTime(
+      processCommits(List(commitId), project, BatchDate(clock), maybeAccessToken).run(SynchronizationSummary())
+    ) flatMap { case (elapsedTime: ElapsedTime, summary) =>
+      logSummary(commitId, project)(elapsedTime, summary._2)
+    }
+
+  private val DontCareCommitId = CommitId("0000000000000000000000000000000000000000")
+
+  private def processCommits(commitList: List[CommitId],
+                             project:    Project,
+                             batchDate:  BatchDate,
+                             maybeToken: Option[AccessToken]
+  ): SummaryState[Interpretation, SynchronizationSummary] =
+    commitList match {
+      case commitId :: commitIds =>
+        StateT
+          .liftF(
+            measureExecutionTime(
+              getInfoFromELandGL(commitId, project, maybeToken)
+                .flatMap {
+                  case (_, Some(CommitInfo(DontCareCommitId, _, _, _, _, _))) =>
+                    Skipped.pure[Interpretation].widen[UpdateResult].map(result => (result, commitId, commitIds))
+                  case (Some(_), Some(_)) =>
+                    Existed.pure[Interpretation].widen[UpdateResult].map(result => (result, commitId, commitIds))
+                  case (None, Some(commitFromGL)) =>
+                    commitToEventLog
```

Comment:
Similarly as above what if we import members of the `commitToEventLog` and `commitEVentsRemover` so we can use their defs directly?